homeassistant/components/unifiprotect/binary_sensor.py

```diff
@@ -77,11 +78,12 @@ class ProtectBinaryEntityDescription(
 
 SENSE_SENSORS: tuple[ProtectBinaryEntityDescription, ...] = (
     ProtectBinaryEntityDescription(
-        key="door",
-        name="Door",
+        key=_KEY_DOOR,
+        name="Contact",
         device_class=BinarySensorDeviceClass.DOOR,
         ufp_value="is_opened",
         ufp_last_trip_value="open_status_changed_at",
+        ufp_enabled="is_contact_sensor_enabled",
     ),
```

Comment:
Does it make sense to have a separate description for each one based on the mount type?

Reply:
I was only able to get updating the device class to work dynamically. Changing the name would not cause an update unless the integration was restarted/reloaded. So, I think just updating the `device_class` dynamically seems to be the easiest way to do it without as much code duplication.

Reply:
It is a bit unexpected that the device class can change here.   I'd normally expect the integration should be reloaded in this case, but that really doesn't get us anything.  Realistically the user is going to set this up once and its probably not going to change unless they repurpose the sensor so I think this is ok.